gaffer-core/graph/src/main/java/gaffer/graph/hook/OperationChainLimiter.java

```diff
@@ -0,0 +1,262 @@
+            if (null != roleScore) {
+                if (roleScore > maxUserScore) {
+                    maxUserScore = roleScore;
+                }
+            }
+        }
+        LOGGER.info("Returning users max operation chain limit score of " + maxUserScore);
+        return maxUserScore;
+    }
+
+    @Override
+    public void postExecute(final Object result, final OperationChain<?> opChain, final User user) {
+        // This method can be overridden to add additional authorisation checks on the results.
+    }
+
+    protected Integer authorise(final Operation operation, final User user) {
+        Integer operationScore = null;
+        if (null != operation) {
+            final Class<? extends Operation> opClass = operation.getClass();
+            for (Map.Entry<Class<? extends Operation>, Integer> entry : operationScores.entrySet()) {
+                if (entry.getKey().isAssignableFrom(opClass)) {
+                       operationScore = entry.getValue();
+                   }
+                }
+            }
+        if (operationScore != null) {
+            return operationScore;
+        } else {
+            LOGGER.warn("The operation '" + operation.getClass().getName() + "' was not found in the config file provided the configured default value of " + DEFAULT_OPERATION_SCORE + " will be used");
+            return DEFAULT_OPERATION_SCORE;
+        }
+    }
+
+    private static Properties readProperties(final Path propFileLocation) {
+        Properties props;
+        if (null != propFileLocation) {
+            try {
+                props = readProperties(Files.newInputStream(propFileLocation, StandardOpenOption.READ));
+            } catch (IOException e) {
+                throw new IllegalArgumentException(e);
+            }
+        } else {
+            props = new Properties();
+        }
+
+        return props;
+    }
+
+    private static Properties readProperties(final InputStream stream) {
+        final Properties props = new Properties();
+        if (null != stream) {
+            try {
+                props.load(stream);
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Failed to load store properties file : " + e.getMessage(), e);
+            } finally {
+                IOUtils.closeQuietly(stream);
+            }
+        }
+        return props;
+    }
+
+    private void loadMapsFromProperties(final Properties operationScorePropertiesFile, final Properties operationAuthorisationScoreLimitPropertiesFile) {
+        Map<Class<? extends Operation>, Integer> opScores = new LinkedHashMap<>();
+        for (String opClassName : operationScorePropertiesFile.stringPropertyNames()) {
+            final Class<? extends Operation> opClass;
+            try {
+                opClass = Class.forName(opClassName).asSubclass(Operation.class);
+            } catch (ClassNotFoundException e) {
+                throw new IllegalArgumentException(e);
```

Comment:
A error message would be helpful, e.g. "An operation class could not be found for operation score property " + opClassName.